src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -92,11 +91,12 @@ data MergeTrace =
       RunBloomFilterAlloc
       MergePolicyForLevel
       Merge.Level
-  | TraceCompletedMerge
+  | -- | This is traced at the latest point the merge could complete. To trace
+    --   at the exact point it really completed, we'd need to do it in the
+    --   @MergingRun@ module.
+    TraceExpectCompletedMerge
```

Comment:
Ah, is that handled correctly by Haddock now? I vague remember it being an issue because the comment is not next to the constructor is is supposed to describe (the `|` is in the way).

Reply:
Seems like this got fixed at some point, verified by adding `{-# OPTIONS_GHC -Winvalid-haddock -haddock #-}`.